File: youtubevideo/src/main/java/com/krake/youtube/YoutubeVideoUtils.java
package com.krake.youtube;

import android.net.Uri;

import com.krake.youtube.model.YoutubeVideo;

import java.util.List;

/**
 * Created by joel on 15/01/16.
 */
public class YoutubeVideoUtils {
    static public String extractVideoIdentifier(YoutubeVideo video) {
        Uri youtubeUri = Uri.parse(video.getVideoUrlValue());
        String identifier = youtubeUri.getQueryParameter("v");
        if (identifier == null) {
            List<String> segments = youtubeUri.getPathSegments();
            return segments.get(segments.size() - 1);
        }

        return identifier;
    }
}

File: youtubevideo/src/main/kotlin/com/krake/youtube/YoutubeVideoUtils.kt
package com.krake.youtube

import android.net.Uri
import com.krake.youtube.model.YoutubeVideo

object YoutubeVideoUtils {
    @JvmStatic
    fun extractVideoIdentifier(video: YoutubeVideo): String {
        val youtubeUri = Uri.parse(video.videoUrlValue)
        val identifier = youtubeUri.getQueryParameter("v")
        if (identifier == null) {
            val segments = youtubeUri.pathSegments
            return segments[segments.size - 1]
        }

        return identifier
    }
}